src/components/navigation/navigation-item/navigation-item.component.ts

```diff
@@ -90,17 +90,22 @@ export class NavigationItemComponent implements AfterViewInit, AfterContentInit,
         this._childrenChanges.unsubscribe();
     }
 
-    private hasActiveLink(link: string | UrlTree): void {
+    private hasActiveLink(link: string | UrlTree): boolean {
+        // If this component has a link, check if it is active.
         if (link) {
-            this.expanded = this._router.isActive(link, true);            
-            if (this.expanded && this._parent) {
-                this._parent.hasActiveLink(link);
+            let isActive = this._router.isActive(link, true);
+            if (isActive) {
+                return true;
             }
+        }
 
-            return;
+        // If this component has children, check if any of them, or their descendants, are active.
+        let hasActiveChildren = false;
+        if (this.children.length > 0) {
+            hasActiveChildren = this.children.some((item) => item.hasActiveLink(item.link));
         }
 
-        this.expanded = false;
+        return hasActiveChildren;            
```

Comment:
It's fine if you want to write this out for your own understanding, but try to condense it and remove redundant variables, since the browser ultimately has to download the javascript version of this.

This would be more readable and concise:

````
if (link && this._router.isActive(link, true)) {
  return true;
}

return this.children.some((item) => item.hasActiveLink(item.link));
````

Reply:
Condensed code as suggested.